plugins/modules/packaging/os/slackpkg.py

```diff
@@ -67,12 +67,13 @@
 
 def query_package(module, slackpkg_path, name):
 
-    import glob
     import platform
+    import os
+    import re
 
     machine = platform.machine()
-    packages = glob.glob("/var/log/packages/%s-*-[%s|noarch]*" % (name,
-                                                                  machine))
+    pattern = re.compile('^%s-[^-]+-(%s|noarch)-[^-]+$' % (name, machine))
```

Comment:
Doesn't it make more sense to escape the stirngs before putting them in?
```suggestion
    pattern = re.compile('^%s-[^-]+-(%s|noarch)-[^-]+$' % (re.escape(name), re.escape(machine)))
```